File: pset6/Vigenere.py
from sys import argv,exit

def main():
    #Check usage
    if len(argv) != 2:
        print("Usage: python Vigenere.py k")
        exit(1)
    elif not argv[1].isalpha():
        print("Usage: python Vigenere.py k")
        exit(1)

    #Define values
    plaintext = input("plaintext: ")
    lengt = len(argv[1])
    ciphertext = ""
    l = 0

    for i in plaintext:

        #Check length of key
        if l == lengt:
            l = 0

        #Check if letter in alphabet
        if i.isalpha():
            #Check digit of plaintext is lowercase or uppercase
            if i.islower():
                ciphertext += chr((ord(i)%97+shift(argv[1][l]))%26+97)
            else:
                ciphertext += chr((ord(i)%65+shift(argv[1][l]))%26+65)
            l+=1
        else:
            ciphertext += i
    #Print cipher text
    print("ciphertext: "+ciphertext)

#Check digit of key is lowercase or uppercase
def shift(c):
    if c.islower():
        return ord(c)%97
    else:
        return ord(c)%65
    

if __name__ == "__main__":
    main()

File: pset2/Vigenere.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <stdlib.h>

int shift(char c);

int main(int argc, string argv[]){

    int keywordLenght;

    //Check usage
    if(argc==2){
        keywordLenght = strlen(argv[1]);

        for(int i=0; i < keywordLenght; i++){
            if(isalpha(argv[1][i]) == 0){
                printf("Usage: ./Vigenere keyword\n");
                return 1;
            }
        }
    }else{
        printf("Usage: ./Vigenere keyword\n");
        return 1;
    }

    //Define values
    string plainText = get_string("plaintext: ");
    int plainTextLenght = strlen(plainText);
    char cipherText[plainTextLenght+1];

    for(int i=0,l=0;i<plainTextLenght;i++){
        //Check end of the key
        if(l == keywordLenght){
            l = 0;
        }

        //Check digit for alphabet
        if(isalpha(plainText[i]) != 0){
            //Check if letter lowercase or uppercase 
            if(islower(plainText[i]) != 0){
                cipherText[i] = (plainText[i]%97+shift(argv[1][l]))%26+97;
            }else{
                cipherText[i] = (plainText[i]%65+shift(argv[1][l]))%26+65;
            }
            //Increase key
            l++;
        }else{
            cipherText[i]=plainText[i];
        }
    }
    // Print cipher text
    printf("ciphertext: ");

    for(int i=0;i<plainTextLenght;i++){
        printf("%c",cipherText[i]);
    }

    printf("\n");
}

//ASCII to decimal
int shift(char c){
    if(islower(c) != 0){
        return c%97;
    }else{
        return c%65;
    }
}
